hashfs/hashfs.py

```diff
@@ -3,11 +3,11 @@
 
 from collections import namedtuple
 from contextlib import contextmanager, closing
-from distutils.dir_util import mkpath
 import glob
 import hashlib
 import io
 import os
+from os.path import exists, isdir
```

Comment:
Nit: I prefer to use `os.path.<object>` directly instead of importing from `os.path`.